src/parser.ts

```diff
@@ -80,6 +80,11 @@ export interface LayerDefinition {
   end: number;
 }
 
+export enum LayerType{
+  VARIABLE_Z,
+  LAYER_COMMENTS,
```

Comment:
```suggestion
  /**
   * Layers are defined by the Z value of the points.
   * This may not be accurate, and may have problems if used with Z-Hops or similar things.
   */
  VARIABLE_Z,

  /**
   * Layer changes are defined by Layer comments in the GCode.
   * This is more accurate but requires the GCode to have layer comments.
   * ;LAYER:0
   */
  LAYER_COMMENTS,
```